mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -72,10 +73,15 @@
     public List<MessageAttachment> retrieveAttachments(InputStream fullContent) throws MimeException, IOException {
         DefaultMessageBuilder defaultMessageBuilder = new DefaultMessageBuilder();
         defaultMessageBuilder.setMimeEntityConfig(MIME_ENTITY_CONFIG);
-        Body body = defaultMessageBuilder
-                .parseMessage(fullContent)
-                .getBody();
+        Message message = defaultMessageBuilder.parseMessage(fullContent);
+        Body body = message.getBody();
         try {
+            Optional<ContentDispositionField> contentDisposition = readHeader(message, CONTENT_DISPOSITION, ContentDispositionField.class);
+
+            if (contentDisposition.isPresent() && contentDisposition.get().isAttachment()) {
```

Comment:
java 6 guys...

I'm OK with optional magic, but I've been bashed not so long ago because of it.

I'll extract the check.